File: Structural/Decorator/Python/Structure/ConcreteDecoratorA.py
"""
ConcreteDecoratorA: Adds responsibilities to the component.
"""
from Decorator import Decorator


class ConcreteDecoratorA(Decorator):
    """Concrete decorator that adds behavior A."""
    
    def operation(self) -> str:
        """Add behavior A before calling the wrapped component."""
        return f"ConcreteDecoratorA({self.component.operation()})"

File: Creational/Decorator/Java/Structure/ConcreteDecoratorA.java
package Creational.Decorator.Java.Structure;

public class ConcreteDecoratorA extends Decorator {
    public ConcreteDecoratorA(Component wrappee) {
        super(wrappee);
    }

    @Override
    public String operation() {
        return wrappee.operation() + " World";
    }
}
